modules/charon-core/src/main/java/org/wso2/charon3/core/config/SCIMSystemSchemaExtensionBuilder.java

```diff
@@ -0,0 +1,163 @@
+     * @throws InternalErrorException   If an error occurred while building the schema.
+     */
+    public void buildSystemSchemaExtension(String configFilePath) throws CharonException, InternalErrorException {
+
+        File provisioningConfig = new File(configFilePath);
+        try (InputStream configFilePathInputStream = new FileInputStream(provisioningConfig)) {
+            buildSystemSchemaExtension(configFilePathInputStream);
+        } catch (FileNotFoundException e) {
+            throw new CharonException(configFilePath + " file not found!", e);
+        } catch (JSONException e) {
+            throw new CharonException("Error while parsing " + configFilePath + " file!", e);
+        } catch (IOException e) {
+            throw new CharonException("Error while closing " + configFilePath + " file!", e);
+        }
+    }
+
+    /**
+     * Build the system schema extension from the input stream.
+     *
+     * @param inputStream               The input stream.
+     * @throws CharonException          If an error occurred while reading the configuration.
+     * @throws InternalErrorException   If an error occurred while building the schema.
+     */
+    public void buildSystemSchemaExtension(InputStream inputStream) throws CharonException, InternalErrorException {
+
+        readConfiguration(inputStream);
+        for (Map.Entry<String, ExtensionAttributeSchemaConfig> attributeSchemaConfig : extensionConfig.entrySet()) {
+            // If there are no children it is a simple attribute, build it.
+            if (!attributeSchemaConfig.getValue().hasChildren()) {
+                buildSimpleAttributeSchema(attributeSchemaConfig.getValue(), attributeSchemas);
+            } else {
+                // Need to build child schemas first.
+                buildComplexAttributeSchema(attributeSchemaConfig.getValue(), attributeSchemas, extensionConfig);
+            }
+        }
+
+        extensionSchema = attributeSchemas.get(extensionRootAttributeURI);
+    }
+
+    /**
+     * Read the configuration from the input stream.
+     *
+     * @param inputStream               The input stream.
+     * @throws CharonException          If an error occurred while reading the configuration.
+     */
+    public void readConfiguration(InputStream inputStream) throws CharonException {
+
+        Scanner scanner = new Scanner(inputStream, "utf-8").useDelimiter("\\A");
+        String jsonString = scanner.hasNext() ? scanner.next() : "";
+
+        JSONArray attributeConfigArray = new JSONArray(jsonString);
+
+        for (int index = 0; index < attributeConfigArray.length(); ++index) {
+            JSONObject rawAttributeConfig = attributeConfigArray.getJSONObject(index);
+            ExtensionAttributeSchemaConfig schemaAttributeConfig =
+                    new ExtensionAttributeSchemaConfig(rawAttributeConfig);
+            if (schemaAttributeConfig.getURI().startsWith(extensionRootAttributeURI)) {
```

Comment:
Can the schemaAttributeConfig.getURI() be null?

Reply:
If the input stream is not null and the JSON content is in the proper format, it will not be null. For extra safety, we can add a null check there.